unit_tests/library/json.cpp

```diff
@@ -27,7 +26,7 @@ BOOST_AUTO_TEST_CASE(test_json_linestring)
     const auto coords = geom.values["coordinates"].get<util::json::Array>().values;
     BOOST_CHECK_EQUAL(coords.size(), 3); // array of three location arrays
 
-    for (const auto each : coords)
+    for (const auto &each : coords)
```

Comment:
Unit test changes are due to compiler errors from updated macOS clang.

Reply:
This crashes clang?

Reply:
Running `make tests` with  clang 12

> error: loop variable 'each' creates a copy from type 'const mapbox::util::variant<osrm::util::json::String, osrm::util::json::Number, mapbox::util::recursive_wrapper<osrm::util::json::Object>, mapbox::util::recursive_wrapper<osrm::util::json::Array>, osrm::util::json::True, osrm::util::json::False, osrm::util::json::Null>' [-Werror,-Wrange-loop-construct]
    for (const auto each : coords)

Reply:
Ah I see `-Werror` strikes again.